test/e2e/app_management_test.go

```diff
@@ -373,6 +373,49 @@ func TestKsonnetApp(t *testing.T) {
 		})
 }
 
+func TestKustomize2AppSource(t *testing.T) {
+
+	patchLabelMatchesFor := func(kind string) func(app *Application) {
+		return func(app *Application) {
+			name := "k2-patched-guestbook-ui"
+			labelValue, err := fixture.Run(
+				"", "kubectl", "-n="+fixture.DeploymentNamespace(),
+				"get", kind, name,
+				"-ojsonpath={.metadata.labels.patched-by}")
+			assert.NoError(t, err)
+			assert.Equal(t, "argo-cd", labelValue, "wrong value of 'patched-by' label of %s %s", kind, name)
+		}
+	}
+
+	Given(t).
+		Path(guestbookPath).
+		NamePrefix("k2-").
+		When().
+		Create().
+		Refresh(RefreshTypeHard).
```

Comment:
minor - don't think you need `Refresh` here?

Reply:
I'm pretty sure I put it there because otherwise the test was failing... Need to check again, though